From 30a602a9a54cb71208670cf907ca4221f7422205 Mon Sep 17 00:00:00 2001
From: Olivier Michel <Olivier.Michel@cyberbotics.com>
Date: Fri, 31 May 2019 15:07:06 +0200
Subject: [PATCH 06/12] Handle spaces in path for default manifest

---
 gcc/gcc.cc | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index 4fd87f2c4a1..449ce4b5ea3 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -7029,7 +7029,6 @@ eval_spec_function (const char *func, const char *args,
   int save_arg_going;
   int save_delete_this_arg;
   int save_this_is_output_file;
-  int save_this_is_library_file;
   int save_input_from_pipe;
   int save_this_is_linker_script;
   const char *save_suffix_subst;
@@ -7047,7 +7046,6 @@ eval_spec_function (const char *func, const char *args,
   save_arg_going = arg_going;
   save_delete_this_arg = delete_this_arg;
   save_this_is_output_file = this_is_output_file;
-  save_this_is_library_file = this_is_library_file;
   save_this_is_linker_script = this_is_linker_script;
   save_input_from_pipe = input_from_pipe;
   save_suffix_subst = suffix_subst;
@@ -7079,13 +7077,15 @@ eval_spec_function (const char *func, const char *args,
 			 argbuf.address ());
 
   /* Pop the spec processing context.  */
+  const char *saved_library_file = this_is_library_file ? argbuf.address()[0] : NULL;
   argbuf.release ();
   argbuf = save_argbuf;
+  if (saved_library_file)
+    argbuf.safe_push (saved_library_file);
 
   arg_going = save_arg_going;
   delete_this_arg = save_delete_this_arg;
   this_is_output_file = save_this_is_output_file;
-  this_is_library_file = save_this_is_library_file;
   this_is_linker_script = save_this_is_linker_script;
   input_from_pipe = save_input_from_pipe;
   suffix_subst = save_suffix_subst;
@@ -7157,8 +7157,9 @@ handle_spec_function (const char *p, bool *retval_nonnull,
   /* p now points to just past the end of the spec function expression.  */
 
   funcval = eval_spec_function (func, args, soft_matched_part);
-  if (funcval != NULL && do_spec_1 (funcval, 0, NULL) < 0)
+  if (funcval != NULL && this_is_library_file == 0 && do_spec_1 (funcval, 0, NULL) < 0)
     p = NULL;
+
   if (retval_nonnull)
     *retval_nonnull = funcval != NULL;
 
-- 
2.51.0.windows.1

